src/braze-upload/lambda/lambda.ts

```diff
@@ -56,7 +56,8 @@ export async function handler(event: Event, context: any): Promise<any> {
 
     const resultPromises = maybeReferralCodes
         .filter(maybeReferralCode => !!maybeReferralCode)
-        .map(async referralCode => {
+        .map(c => c as string)  // typescript doesn't know that the above line filters to strings only
```

Comment:
You could delete this line if you make the above line:
```suggestion
        filter(maybeReferralCode => String.apply(!!maybeReferralCode))
```

Reply:
unfortunately typescript still considers that a `string | null` without the cast

Reply:
I think the weakness is in the `filter` definition, where the return type is the same:
`filter(callbackfn: (value: T, index: number, array: T[]) => any, thisArg?: any): T[];`

I've been meaning to try the fp-ts library (which seems quite scala-influenced), which has an [Array type](https://gcanti.github.io/fp-ts/modules/Array.ts.html) which does what we need:
`const filterMap: <A, B>(f: (a: A) => Option<B>) => (fa: A[]) => B[]`
I'll leave that for a future refactor...

Reply:
That's weird the `| null`, given this:

![Screen Shot 2020-09-24 at 08 21 36](https://user-images.githubusercontent.com/1515970/94113591-08aa6180-fe3f-11ea-8174-47a2cd1fc30f.png)

Reply:
The original array has type `(string | null)[]`, and filter doesn't allow you to change the type to `string[]` - there has to be a `map` as well for that

Reply:
Gotcha. Using `.map` is better.